packages/perspective-viewer-d3fc/src/js/axis/crossAxis.js

```diff
@@ -46,13 +46,19 @@ export const domain = settings => {
     let valueName = "crossValue";
     let settingName = "crossValues";
 
-    const extentTime = fc.extentTime().accessors([d => new Date(d[valueName])]);
+    const extentTime = fc
+        .extentTime()
+        .accessors([d => new Date(d[valueName])])
+        .padUnit("domain");
 
     const _domain = function(data) {
         const flattenedData = flattenArray(data);
         switch (axisType(settings, settingName)) {
             case AXIS_TYPES.time:
-                return extentTime(flattenedData);
+                const dataStart = flattenedData[0][valueName];
+                const dataNext = flattenedData.find(d => d[valueName] !== dataStart);

```

Comment:
@DevAndyLee, can you think of a better way to do this? I.e. find the date value of the next bar. This doesn't feel like a very robust method, unless we rely on the data always being ordered correctly.